### PR TITLE
fix: show base-url warning in launchpad

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes [UNIFY-1226](https://cypress-io.atlassian.net/browse/UNIFY-1226)

### User facing changelog
Show base-url warning in launchpad if Cypress cannot connect to provided baseUrl.

### Additional details
Adding the baseUrl warning + retry from 9.x into 10.x. There was a styling bug in the `Warning.vue` component so that was fixed as well.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/157124668-d98673a2-cb83-4890-89a4-06550dae4e7d.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by JessicaSachs** [comment]
@ZachJW34 does that match the mocks? Our Alert component generally has a yellow header.

**Comment by ZachJW34** [comment]
@JessicaSachs there was a bug in the Warning component that was causing the default styles to drop out so I've fixed that.

**Review by elevatebart — COMMENTED** [review]
A few aestetic changes, overall great

**Review by elevatebart — DISMISSED** [review]
I made the changes to your brnach. I hope it's to your liking.

**Comment by elevatebart** [comment]
@ZachJW34 looks like this now ☺️
Screen Shot 2022-03-08 at 1 09 07 PM

File: packages/frontend-shared/src/locales/en-US.json [hunk 1]
```diff
@@ -643,5 +643,8 @@
       "invalidSelector": "Invalid",
       "selectorMethodsLabel": "Selector Methods"
     }
+  },
+  "warnings": {
+    "retry": "Retry"
```
Comment (elevatebart) [inline packages/frontend-shared/src/locales/en-US.json:648]: I believe that is what the button looks like in the comps.

```suggestion
    "retry": "Try again"
```

**Comment by ZachJW34** [comment]
@elevatebart looks great! Couldn't find any comps so I winged it. Looks like some tests are failing, I'll fix them up

**Review by elevatebart — DISMISSED** [review]

File: packages/data-context/src/actions/ProjectActions.ts [hunk 2]
```diff
@@ -511,4 +513,20 @@ export class ProjectActions {
       }
     })
   }
+
+  async pingBaseUrl () {
+    const baseUrl = (await this.ctx.project.getConfig())?.baseUrl
+
+    // Should never happen
+    if (!baseUrl) {
+      throw new Error('Cannot ping url without a valid baseUrl')
```
Comment (ryanthemanuel) [inline packages/data-context/src/actions/ProjectActions.ts:522]: Out of curiosity, what ends up happening in the app if this error is thrown?

Comment (ZachJW34) [reply packages/data-context/src/actions/ProjectActions.ts:522]: Tested and the error was getting swallowed, a noop makes sense.

Comment (ZachJW34) [reply packages/data-context/src/actions/ProjectActions.ts]: Done in fa94e7e98e3ab17e1a46eb7687b88ee27321a9d2

**Review by elevatebart — APPROVED** [review]
We need to do something about this icon colors classes. The default color does not match the mocks. Neither does the close button. 

Work for another PR I guess.

**Review by ryanthemanuel — APPROVED** [review]

File: packages/launchpad/cypress/e2e/base-url.cy.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+describe('baseUrl', () => {
```
Comment (emilyrohrbough) [inline packages/launchpad/cypress/e2e/base-url.cy.ts:1]: Nit: can we add some context describe how this is specific to the e2e testing type?

**Pull request merged**